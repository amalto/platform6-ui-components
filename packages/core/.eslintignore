dist/
loader/

src/components.d.ts