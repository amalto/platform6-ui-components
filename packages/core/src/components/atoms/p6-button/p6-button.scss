//@import "src/global/_variable";
//@import "src/global/_colors";
//@import "src/global/_functions";
//@import "src/global/_control";

$button-color: $text-strong !default;
$button-background-color: $scheme-main !default;
$button-family: false !default;

$button-border-color: $border !default;
$button-border-width: $control-border-width !default;

$button-padding-vertical: calc(0.5em - #{$button-border-width}) !default;
$button-padding-horizontal: 1em !default;

$button-hover-color: $link-hover !default;
$button-hover-border-color: $link-hover-border !default;

$button-focus-color: $link-focus !default;
$button-focus-border-color: $link-focus-border !default;
$button-focus-box-shadow-size: 0 0 0 0.125em !default;
$button-focus-box-shadow-color: bulmaRgba($link, 0.25) !default;

$button-active-color: $link-active !default;
$button-active-border-color: $link-active-border !default;

$button-text-color: $text !default;
$button-text-decoration: underline !default;
$button-text-hover-background-color: $background !default;
$button-text-hover-color: $text-strong !default;

$button-disabled-background-color: $scheme-main !default;
$button-disabled-border-color: $border !default;
$button-disabled-shadow: none !default;
$button-disabled-opacity: 0.5 !default;

$button-static-color: $text-light !default;
$button-static-background-color: $scheme-main-ter !default;
$button-static-border-color: $border !default;

@import 'src/styles/atoms/control';
@import 'src/styles/atoms/button';

:host {
  display: inline-flex;
  .button {
    @include control;
    @include unselectable;

    background-color: $button-background-color;
    border-color: $button-border-color;
    border-width: $button-border-width;
    color: $button-color;
    cursor: pointer;
    @if $button-family {
      font-family: $button-family;
    }

    justify-content: center;
    padding: $button-padding-vertical $button-padding-horizontal;
    text-align: center;
    white-space: nowrap;

    strong {
      color: inherit;
    }

    ::slotted(.icon:first-child:not(:last-child)),
    ::slotted(p6-icon:first-child:not(:last-child)) {
      margin-left: calc(#{-1 / 2 * $button-padding-horizontal} - #{$button-border-width});
      margin-right: $button-padding-horizontal / 4;
    }

    ::slotted(.icon:last-child:not(:first-child)),
    ::slotted(p6-icon:last-child:not(:first-child)) {
      margin-left: $button-padding-horizontal / 4;
      margin-right: calc(#{-1 / 2 * $button-padding-horizontal} - #{$button-border-width});
    }

    ::slotted(.icon:first-child:last-child),
    ::slotted(p6-icon:first-child:last-child) {
      $margin: calc(#{-1 / 2 * $button-padding-horizontal} - #{$button-border-width});

      margin-left: $margin;
      margin-right: $margin;
    }

    // States
    &:hover,
    &.is-hovered {
      border-color: $button-hover-border-color;
      color: $button-hover-color;
    }

    &:active,
    &.is-active {
      border-color: $button-active-border-color;
      color: $button-active-color;
    }

    &:focus,
    &.is-focused {
      border-color: $button-focus-border-color;
      color: $button-focus-color;
    }

    fieldset[disabled] & {
      background-color: $button-disabled-background-color;
      border-color: $button-disabled-border-color;
      box-shadow: $button-disabled-shadow;
      opacity: $button-disabled-opacity;
    }

    &:focus:not(:active),
    &.is-focused:not(:active) {
      box-shadow: $button-focus-box-shadow-size $button-focus-box-shadow-color;
    }

    // Colors
    &.is-text {
      background-color: transparent;
      border-color: transparent;
      color: $button-text-color;
      text-decoration: $button-text-decoration;

      &:hover,
      &.is-hovered,
      &:focus,
      &.is-focused {
        background-color: $button-text-hover-background-color;
        color: $button-text-hover-color;
      }

      &:active,
      &.is-active {
        background-color: bulmaDarken($button-text-hover-background-color, 5%);
        color: $button-text-hover-color;
      }

      &[disabled],
      fieldset[disabled] & {
        background-color: transparent;
        border-color: transparent;
        box-shadow: none;
      }
    }

    @each $name, $pair in $colors {
      $color: nth($pair, 1);
      $color-invert: nth($pair, 2);
      &.is-#{$name} {
        background-color: $color;
        border-color: transparent;
        color: $color-invert;

        &:hover,
        &.is-hovered {
          background-color: bulmaDarken($color, 2.5%);
          border-color: transparent;
          color: $color-invert;
        }

        &:focus,
        &.is-focused {
          border-color: transparent;
          color: $color-invert;

          &:not(:active) {
            box-shadow: $button-focus-box-shadow-size bulmaRgba($color, 0.25);
          }
        }

        &:active,
        &.is-active {
          background-color: bulmaDarken($color, 5%);
          border-color: transparent;
          color: $color-invert;
        }

        &[disabled],
        fieldset[disabled] & {
          background-color: $color;
          border-color: transparent;
          box-shadow: none;
        }

        &.is-inverted {
          background-color: $color-invert;
          color: $color;

          &:hover,
          &.is-hovered {
            background-color: bulmaDarken($color-invert, 5%);
          }

          &[disabled],
          fieldset[disabled] & {
            background-color: $color-invert;
            border-color: transparent;
            box-shadow: none;
            color: $color;
          }
        }

        &.is-loading {
          &::after {
            border-color: transparent transparent $color-invert $color-invert !important;
          }
        }

        &.is-outlined {
          background-color: transparent;
          border-color: $color;
          color: $color;

          &:hover,
          &.is-hovered,
          &:focus,
          &.is-focused {
            background-color: $color;
            border-color: $color;
            color: $color-invert;
          }

          &.is-loading {
            &::after {
              border-color: transparent transparent $color $color !important;
            }

            &:hover,
            &.is-hovered,
            &:focus,
            &.is-focused {
              &::after {
                border-color: transparent transparent $color-invert $color-invert !important;
              }
            }
          }

          &[disabled],
          fieldset[disabled] & {
            background-color: transparent;
            border-color: $color;
            box-shadow: none;
            color: $color;
          }
        }

        &.is-inverted.is-outlined {
          background-color: transparent;
          border-color: $color-invert;
          color: $color-invert;

          &:hover,
          &.is-hovered,
          &:focus,
          &.is-focused {
            background-color: $color-invert;
            color: $color;
          }

          &.is-loading {
            &:hover,
            &.is-hovered,
            &:focus,
            &.is-focused {
              &::after {
                border-color: transparent transparent $color $color !important;
              }
            }
          }

          &[disabled],
          fieldset[disabled] & {
            background-color: transparent;
            border-color: $color-invert;
            box-shadow: none;
            color: $color-invert;
          }
        }

        // If light and dark colors are provided
        @if length($pair) >= 4 {
          $color-light: nth($pair, 3);
          $color-dark: nth($pair, 4);
          &.is-light {
            background-color: $color-light;
            color: $color-dark;

            &:hover,
            &.is-hovered {
              background-color: bulmaDarken($color-light, 2.5%);
              border-color: transparent;
              color: $color-dark;
            }

            &:active,
            &.is-active {
              background-color: bulmaDarken($color-light, 5%);
              border-color: transparent;
              color: $color-dark;
            }
          }
        }
      }
    }
    // Sizes
    &.is-small {
      @include button-small;
    }

    &.is-normal {
      @include button-normal;
    }

    // Modifiers
    &[disabled] {
      background-color: $button-disabled-background-color;
      border-color: $button-disabled-border-color;
      box-shadow: $button-disabled-shadow;
      opacity: $button-disabled-opacity;
    }

    &.is-fullwidth {
      display: flex;
      width: 100%;
    }

    &.is-loading {
      color: transparent !important;
      pointer-events: none;

      &::after {
        @include loader;
        @include center(1em);

        position: absolute !important;
      }
    }

    &.is-static {
      background-color: $button-static-background-color;
      border-color: $button-static-border-color;
      color: $button-static-color;
      box-shadow: none;
      pointer-events: none;
    }

    &.is-rounded {
      border-radius: $radius-rounded;
      padding-left: calc(#{$button-padding-horizontal} + 0.25em);
      padding-right: calc(#{$button-padding-horizontal} + 0.25em);
    }
  }
}
