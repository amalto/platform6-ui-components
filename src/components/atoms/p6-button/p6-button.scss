@import "bulma/sass/elements/button";

:host {
  display: inline-block;

  button {
    width: 100%;
  }
}
