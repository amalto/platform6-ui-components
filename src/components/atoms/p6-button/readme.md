# p6-button

<!-- Auto Generated Below -->

## Properties

| Property   | Attribute  | Description                                                     | Type                                                                     | Default     |
| ---------- | ---------- | --------------------------------------------------------------- | ------------------------------------------------------------------------ | ----------- |
| `disabled` | `disabled` | Disabled - If `true`, the user cannot interact with the button. | `boolean`                                                                | `false`     |
| `mode`     | `mode`     | set the mode of the button                                      | `"danger" \| "default" \| "info" \| "primary" \| "success" \| "warning"` | `"default"` |
| `outlined` | `outlined` | Outlined                                                        | `boolean`                                                                | `false`     |
| `size`     | `size`     | set the size of the button                                      | `"default" \| "large" \| "medium" \| "normal" \| "small"`                | `"default"` |
| `type`     | `type`     | type of the button.                                             | `"button" \| "reset" \| "submit"`                                        | `"submit"`  |
| `waiting`  | `waiting`  | If set, shows a waiting/busy indicator                          | `boolean`                                                                | `false`     |

---

_Built with [StencilJS](https://stenciljs.com/)_
