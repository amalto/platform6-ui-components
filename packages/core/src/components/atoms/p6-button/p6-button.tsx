import { Component, ComponentInterface, Element, h, JSX, Prop } from '@stencil/core';
import { Mode, Size } from '../../../shared/types';
import { getModeClass, getSizeClass } from '../../../utils/classes';

export type P6ButtonType = 'submit' | 'reset' | 'button';

@Component({
  tag: 'p6-button',
  styleUrl: 'p6-button.scss',
  shadow: true,
})
export class P6Button implements ComponentInterface {
  @Element() host!: HTMLP6ButtonElement;

  /**
   * set the mode of the button
   */
  @Prop() mode: Mode = Mode.default;

  /**
   * Outlined
   */
  @Prop() outlined = false;

  /**
   * If set, shows a waiting/busy indicator
   */
  @Prop() waiting = false;

  /**
   * set the size of the button
   */
  @Prop() size: Size = Size.normal;

  /**
   * type of the button.
   */
  @Prop() type: P6ButtonType = 'submit';

  /**
   * Disabled - If `true`, the user cannot interact with the button.
   */
  @Prop() disabled = false;

  private clickDelegationHandler = (): void => {
    if (this.type === 'submit' || this.type === 'reset') {
      const form = this.host?.closest('form');
      if (form !== null && form !== undefined) {
        const nativeButton = document.createElement('button');
        nativeButton.type = this.type;
        nativeButton.style.display = 'none';
        form.appendChild(nativeButton);
        nativeButton.click();
        form.removeChild(nativeButton);
      }
    }
  };

  render(): JSX.Element {
    const classes = {
      'button': true,
      ...getModeClass(this.mode),
      ...getSizeClass(this.size),
      'is-outlined': this.outlined,
      'is-loading': this.waiting,
    };

    return (
      <button
        class={classes}
        // eslint-disable-next-line react/button-has-type
        type={this.type}
        disabled={this.disabled}
        aria-disabled={this.disabled ? 'true' : null}
        onClick={this.clickDelegationHandler}
      >
        <slot />
      </button>
    );
  }
}
