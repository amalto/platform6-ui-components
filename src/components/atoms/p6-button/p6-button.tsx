import { Component, Element, h, Prop } from "@stencil/core";
import { Mode, Size } from "~shared/types";
import { isEmpty } from "~utils/attribute";

export type P6ButtonType = "submit" | "reset" | "button";

@Component({
  tag: "p6-button",
  styleUrl: "p6-button.scss",
  shadow: true,
})
export class P6Button {
  @Element() host?: HTMLP6ButtonElement;

  /**
   * set the mode of the button
   */
  @Prop() mode: Mode = "default";

  /**
   * Outlined
   */
  @Prop() outlined = false;

  /**
   * If set, shows a waiting/busy indicator
   */
  @Prop() waiting = false;

  /**
   * set the size of the button
   */
  @Prop() size: Size = "default";

  /**
   * type of the button.
   */
  @Prop() type: P6ButtonType = "submit";

  /**
   * Disabled - If `true`, the user cannot interact with the button.
   */
  @Prop() disabled = false;

  private clickDelegationHandler = (): void => {
    if (this.type === "submit" || this.type === "reset") {
      const form = this.host?.closest("form");
      if (form !== null && form !== undefined) {
        const nativeButton = document.createElement("button");
        nativeButton.type = this.type;
        nativeButton.style.display = "none";
        form.appendChild(nativeButton);
        nativeButton.click();
        form.removeChild(nativeButton);
      }
    }
  };

  render(): JSX.Element {
    const classes = {
      button: true,
      [`is-${this.mode}`]: !isEmpty(this.mode) && this.mode !== "default",
      [`is-${this.size}`]: !isEmpty(this.size) && this.size !== "default",
      "is-outlined": !!this.outlined,
      "is-loading": !!this.waiting,
    };

    return (
      <button
        class={classes}
        // eslint-disable-next-line react/button-has-type
        type={this.type}
        disabled={this.disabled}
        aria-disabled={this.disabled ? "true" : null}
        onClick={this.clickDelegationHandler}
      >
        <slot />
      </button>
    );
  }
}
