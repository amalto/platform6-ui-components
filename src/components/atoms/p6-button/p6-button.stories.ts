import { Components } from '../../../components';
import { ComponentProps, getElement, getSelectArgType, makeModeStory, makeSizeStory, makeStory, Props, StringSelectArgType } from '../../../shared/storybook';
import { Mode, Size } from '../../../shared/types';
import { P6ButtonType } from './p6-button';

const component = 'p6-button';

export default {
  title: 'Atoms/Button',
  component,
};

const componentProps: ComponentProps = ['mode', 'outlined', 'waiting', 'size', 'type', 'disabled'];

const getStoryField = (text: string | HTMLElement[], props?: Props<Components.P6Button>): HTMLElement => {
  return getElement(component, text, props);
};

export const Default = makeStory<{
  text: string;
  size: Size;
  mode: Mode;
  disabled: boolean;
  outlined: boolean;
  waiting: boolean;
  type: P6ButtonType;
}>({
  componentProps,
  args: {
    text: 'My button',
    type: 'button',
    size: Size.normal,
    mode: Mode.default,
    disabled: false,
    outlined: false,
    waiting: false,
  },
  argTypes: {
    ...getSelectArgType<StringSelectArgType>(
      'type',
      ['submit', 'reset', 'button'].map(type => ({ key: type, value: type })),
    ),
  },
  builder: ({ text, ...args }) => getStoryField(text, { ...args }),
});

export const Sizes = makeSizeStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField(key, {
      size: value,
    }),
});

export const Modes = makeModeStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField(key, {
      mode: value,
    }),
});

export const Icon = makeStory({
  componentProps,
  builder: (): HTMLElement =>
    getElement('div', [
      getStoryField(
        [getElement('p6-icon', [], { name: 'home', size: Size.small }), getElement('span', '')],
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        { size: Size.small, style: 'margin-right: 5px;' as any },
      ),
      getStoryField([
        getElement('p6-icon', [], {
          name: 'home',
          style: 'margin-right: 5px;',
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
        } as any),
        getElement('span', 'Home'),
      ]),
    ]),
});

export const Waiting = makeStory<{ waiting: boolean }>({
  componentProps,
  args: {
    waiting: true,
  },
  builder: (props): HTMLElement => getStoryField('Waiting', props),
});

export const Disabled = makeStory<{ disabled: boolean }>({
  componentProps,
  args: {
    disabled: true,
  },
  builder: (props): HTMLElement => getStoryField('Disabled', props),
});

export const Outlined = makeStory<{ outlined: boolean }>({
  componentProps,
  args: {
    outlined: true,
  },
  builder: (props): HTMLElement => getStoryField('Outlined', props),
});
