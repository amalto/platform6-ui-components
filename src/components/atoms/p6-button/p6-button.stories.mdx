import { library } from "@fortawesome/fontawesome-svg-core";
import { fab } from "@fortawesome/free-brands-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { action } from "@storybook/addon-actions";
import { Meta, Preview, Story } from "@storybook/addon-docs/blocks";
import { select, text, boolean, withKnobs } from "@storybook/addon-knobs";
import readme from "./readme.md";

<Meta
  title="Atoms/Button"
  component="p6-button"
  parameters={{ notes: readme }}
/>

# Button

<Story
  name="Default"
  parameters={{ docs: { disable: true } }}
  decorators={[withKnobs]}
>
  {(el) => {
    const modes = [
      "danger",
      "warning",
      "default",
      "info",
      "success",
      "primary",
    ];
    const sizes = ["small", "default", "normal", "medium", "large"];
    const kModes = select("Modes", modes, "default");
    const kSize = select("Size", sizes, "default");
    const kOutlined = boolean("Outlined", false);
    const kWaiting = boolean("Waiting", false);
    const kDisabled = boolean("Disabled", false);
    const component = document.createElement("p6-button");
    component.mode = kModes;
    component.waiting = kWaiting;
    component.disabled = kDisabled;
    component.size = kSize;
    component.outlined = kOutlined;
    component.addEventListener("click", action("Click"));
    component.innerHTML = `Click!`;
    return component;
  }}
</Story>

<Preview>
  <Story name="Size" height="42px">
    {`
      <p6-button size="small">small</p6-button>
      <p6-button>default</p6-button>
      <p6-button size="normal">normal</p6-icon></p6-button>
      <p6-button size="medium">medium</p6-button>
      <p6-button size="large">large</p6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="Mode" height="42px">
    {`
      <P6-button mode="danger">danger</P6-button>
      <P6-button mode="warning">warning</P6-button>
      <P6-button mode="default">default</P6-button>
      <P6-button mode="info">info</P6-button>
      <P6-button mode="success">success</P6-button>
      <P6-button mode="primary">primary</P6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="Waiting" height="42px">
    {`
      <P6-button mode="danger" waiting>danger</P6-button>
      <P6-button mode="warning" waiting>warning</P6-button>
      <P6-button mode="default" waiting>default</P6-button>
      <P6-button mode="info" waiting>info</P6-button>
      <P6-button mode="success" waiting>success</P6-button>
      <P6-button mode="primary" waiting>primary</P6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="Disabled" height="42px">
    {`
      <P6-button mode="danger" disabled>danger</P6-button>
      <P6-button mode="warning" disabled>warning</P6-button>
      <P6-button mode="default" disabled>default</P6-button>
      <P6-button mode="info" disabled>info</P6-button>
      <P6-button mode="success" disabled>success</P6-button>
      <P6-button mode="primary" disabled>primary</P6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="Outlined" height="42px">
    {`
      <P6-button mode="danger" outlined>danger</P6-button>
      <P6-button mode="warning" outlined>warning</P6-button>
      <P6-button mode="default" outlined>default</P6-button>
      <P6-button mode="info" outlined>info</P6-button>
      <P6-button mode="success" outlined>success</P6-button>
      <P6-button mode="primary" outlined>primary</P6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="Outlined and Waiting" height="42px">
    {`
      <P6-button mode="danger" outlined waiting>danger</P6-button>
      <P6-button mode="warning" outlined waiting>warning</P6-button>
      <P6-button mode="default" outlined waiting>default</P6-button>
      <P6-button mode="info" outlined waiting>info</P6-button>
      <P6-button mode="success" outlined waiting>success</P6-button>
      <P6-button mode="primary" outlined waiting>primary</P6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="Outlined and Disabled" height="42px">
    {`
      <P6-button mode="danger" outlined disabled>danger</P6-button>
      <P6-button mode="warning" outlined disabled>warning</P6-button>
      <P6-button mode="default" outlined disabled>default</P6-button>
      <P6-button mode="info" outlined disabled>info</P6-button>
      <P6-button mode="success" outlined disabled>success</P6-button>
      <P6-button mode="primary" outlined disabled>primary</P6-button>
    `}
  </Story>
</Preview>

<Preview>
  <Story name="With Icon" height="42px">
    {`
      <p6-button><p6-icon name="home" /></p6-button>
      <p6-button><p6-icon name="home" /></p6-icon> Home</p6-button>
    `}
  </Story>
</Preview>
