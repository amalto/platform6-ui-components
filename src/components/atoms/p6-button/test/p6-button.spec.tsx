import { newSpecPage } from "@stencil/core/testing";
import { P6Button } from "../p6-button";

describe("p6-button", () => {
  describe("mode", () => {
    const modes = ["danger", "warning", "info", "success", "primary"];

    it.each(modes)('has the class "is-%s" when mode is "%s"', async (mode) => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button mode=${mode}></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it("does not add class when mode is default", async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button mode="default"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it("does not add class when mode is undefined", async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button mode=${undefined}></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });

  describe("outlined", () => {
    it('has a "is-outlined" class when outlined is true', async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button outlined="true"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
    it('has no "is-outlined" class when outlined is false', async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button outlined="false"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });

  describe("waiting", () => {
    it('has a "is-loading" class when waiting is true', async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button waiting="true"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
    it('has no "is-loading" class when waiting is false', async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button waiting="false"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });

  describe("disabled", () => {
    it('has a "disabled" attribute on the button when disabled is true', async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button disabled="true"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
    it('has no "disabled" attribute on the button when disabled is false', async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button disabled="false"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });

  describe("size", () => {
    const sizes = ["small", "normal", "medium", "large"];

    it.each(sizes)('has the class "is-%s" when size is "%s"', async (size) => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button size=${size}></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it("does not add class when size is default", async () => {
      const page = await newSpecPage({
        components: [P6Button],
        html: `<p6-button size="default"></p6-button>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });
});
