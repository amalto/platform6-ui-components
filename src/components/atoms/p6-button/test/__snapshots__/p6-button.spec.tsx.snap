// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-button disabled has a "disabled" attribute on the button when disabled is true 1`] = `
<p6-button disabled="true">
  <mock:shadow-root>
    <button aria-disabled="true" class="button" disabled="" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button disabled has no "disabled" attribute on the button when disabled is false 1`] = `
<p6-button disabled="false">
  <mock:shadow-root>
    <button class="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode does not add class when mode is default 1`] = `
<p6-button mode="default">
  <mock:shadow-root>
    <button class="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode does not add class when mode is undefined 1`] = `
<p6-button mode="undefined">
  <mock:shadow-root>
    <button class="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode has the class "is-danger" when mode is "%s" 1`] = `
<p6-button mode="danger">
  <mock:shadow-root>
    <button class="button is-danger" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode has the class "is-info" when mode is "%s" 1`] = `
<p6-button mode="info">
  <mock:shadow-root>
    <button class="button is-info" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode has the class "is-primary" when mode is "%s" 1`] = `
<p6-button mode="primary">
  <mock:shadow-root>
    <button class="button is-primary" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode has the class "is-success" when mode is "%s" 1`] = `
<p6-button mode="success">
  <mock:shadow-root>
    <button class="button is-success" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button mode has the class "is-warning" when mode is "%s" 1`] = `
<p6-button mode="warning">
  <mock:shadow-root>
    <button class="button is-warning" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button outlined has a "is-outlined" class when outlined is true 1`] = `
<p6-button outlined="true">
  <mock:shadow-root>
    <button class="button is-outlined" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button outlined has no "is-outlined" class when outlined is false 1`] = `
<p6-button outlined="false">
  <mock:shadow-root>
    <button class="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button size does not add class when size is default 1`] = `
<p6-button size="default">
  <mock:shadow-root>
    <button class="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button size has the class "is-large" when size is "%s" 1`] = `
<p6-button size="large">
  <mock:shadow-root>
    <button class="button is-large" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button size has the class "is-medium" when size is "%s" 1`] = `
<p6-button size="medium">
  <mock:shadow-root>
    <button class="button is-medium" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button size has the class "is-normal" when size is "%s" 1`] = `
<p6-button size="normal">
  <mock:shadow-root>
    <button class="button is-normal" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button size has the class "is-small" when size is "%s" 1`] = `
<p6-button size="small">
  <mock:shadow-root>
    <button class="button is-small" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button waiting has a "is-loading" class when waiting is true 1`] = `
<p6-button waiting="true">
  <mock:shadow-root>
    <button class="button is-loading" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;

exports[`p6-button waiting has no "is-loading" class when waiting is false 1`] = `
<p6-button waiting="false">
  <mock:shadow-root>
    <button class="button" type="submit">
      <slot></slot>
    </button>
  </mock:shadow-root>
</p6-button>
`;
